### PR TITLE
Feature/tg 95 create classroom

### PR DESCRIPTION
**Pull request by carlos-canedo-dev**
# Summary

Implement create classroom functionality

# Details

- Implement classroom form
- Implement dynamic schedule creation in classroom form
- Update app layout main container scrolling behaviour
- Implement custom time picker
- Implement date mapper
- Implement time mapper
- Update classroom models
- Update list of classrooms component to access new functionality
- Implement create classroom request

# Evidence

feature-tg-95-create-classroom-fe-workflow
feature-tg-95-create-classroom-fe-create-schedule

### CODE REVIEW

File: src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  }
+
+  fetchProgramOptions() {
+    this.programService
+      .getAll()
+      .pipe(
+        map((programs): SelectOption[] =>
+          programs.map(program => ({
+            label: program.name,
+            value: program.id.toString(),
```
Comment (AJN-jalaU) [inline src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts]: we can implement destructuring  in program variable

**Review by AJN-jalaU — APPROVED** [review]

File: src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+  get schedules(): FormArray {
+    return this.classroomForm.get('schedules') as FormArray<ScheduleForm>;
+  }
+
+  addSchedule() {
+    const schedule: ScheduleForm = this.formBuilder.group({
+      weekday: ['', [Validators.required]],
+      startingTime: ['', [Validators.required]],
+      endingTime: ['', [Validators.required]],
```
Comment (RSarabiaJala) [inline src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts]: Add validator for ending time can't be prior to starting time and viceversa
image

**Review by RSarabiaJala — CHANGES_REQUESTED** [review]

File: src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts (hunk 2 above)
Comment (carlos-canedo-dev) [reply src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts]: Updated in last commit

feature-tg-95-create-classroom-fe-create-schedule

File: src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts (hunk 1 above)
Comment (carlos-canedo-dev) [reply src/app/modules/app/pages/classrooms/components/classroom-form/classroom-form.component.ts]: Updated in last commit

**Review by RSarabiaJala — APPROVED** [review]

**Pull request merged**